File: Accupay.Data/Services/Imports/ShiftScheduleRowRecord.cs
using AccuPay.Infrastracture.Services.Excel;
using System;

namespace AccuPay.Data.Services.Imports
{
    public class ShiftScheduleRowRecord : IExcelRowRecord
    {
        [ColumnName("Employee No")]
        public string EmployeeNo { get; set; }

        [ColumnName("Effective Date From")]
        public DateTime StartDate { get; set; }

        [ColumnName("Effective Date To (Optional)")]
        public DateTime? EndDate { get; set; }

        [ColumnName("Time From")]
        public TimeSpan? StartTime { get; set; }

        [ColumnName("Time To")]
        public TimeSpan? EndTime { get; set; }

        [ColumnName("Break Start Time (Optional)")]
        public TimeSpan? BreakStartTime { get; set; }

        [ColumnName("Break Length (Optional)")]
        public decimal BreakLength { get; set; }

        [ColumnName("Offset (Optional) (true/false)")]
        public bool IsRestDay { get; set; }

        public int LineNumber { get; set; }
    }
}

File: AccuPay/Views/Time Attendance Form/Shift Schedule/ShiftScheduleRowRecord.vb
Option Strict On

Imports AccuPay.Data.Interfaces.Excel
Imports AccuPay.Utilities.Attributes

Namespace Global.Globagility.AccuPay.ShiftSchedules

    Public Class ShiftScheduleRowRecord
        Implements IExcelRowRecord

        <ColumnName("Employee No")>
        Public Property EmployeeNo As String

        <ColumnName("Effective Date From")>
        Public Property StartDate As Date

        <ColumnName("Effective Date To (Optional)")>
        Public Property EndDate As Date?

        <ColumnName("Time From")>
        Public Property StartTime As TimeSpan?

        <ColumnName("Time To")>
        Public Property EndTime As TimeSpan?

        <ColumnName("Break Start Time (Optional)")>
        Public Property BreakStartTime As TimeSpan?

        <ColumnName("Break Length (Optional)")>
        Public Property BreakLength As Decimal

        <ColumnName("Offset (Optional) (true/false)")>
        Public Property IsRestDay As Boolean

        Public Property LineNumber As Integer Implements IExcelRowRecord.LineNumber
    End Class

End Namespace